src/jira/util/jwt.ts

```diff
@@ -28,18 +29,22 @@ export enum TokenType {
 	context = "context"
 }
 
-export function extractJwtFromRequest(req: Request): string | undefined {
-	const tokenInQuery = req.query?.[JWT_PARAM];
+let secureJiraHostInCookies;
```

Comment:
This var will bleed between different requests

Reply:
That's what we want :) Once the FF is on, the value will eventually be updated (2nd call after the flip). Given the distributed nature of feature-flags, this is OK imho

Reply:
Yup, understand that, and the full story is:
1. We want to have some feature flags to guard this code. But the middleware is NOT async at the moment, therefore can't use the feature flag (unless we make it async)
2. I understand that making the middleware async should not have problem. But I also think `turning middleware async` vs `have a global var` here, the latter one is less risky.
3. I understand the race condition of multi request, but I think they will all lead to expected result (which is "the flag will be used eventually" if turned on. 

Unless you think there's still impact on this global var?